packages/components/src/sites-table/use-sites-table-sorting.tsx

```diff
@@ -0,0 +1,35 @@
+// eslint-disable-next-line no-restricted-imports
+import type { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+
+interface SitesTableSortOptions {
+	sort?: string;
+	order?: string;
+}
+
+interface UseSitesTableSortingResult {
+	sortedSites: SiteExcerptData[];
+}
+
+export function useSitesTableSorting(
+	allSites: SiteExcerptData[],
+	{ sort, order = 'asc' }: SitesTableSortOptions
+): UseSitesTableSortingResult {
+	switch ( sort ) {
+		case 'updated-at':
+			return { sortedSites: sortSitesByLastPublish( allSites, order ) };
+		default:
+			return { sortedSites: allSites };
+	}
```

Comment:
Maybe we should wrap this in a `useMemo` so that we only resort when it's actually necessary.